marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/flow/stepExecutionContext.sjs

```diff
@@ -175,7 +175,7 @@ class StepExecutionContext {
 
         // Added in 5.5 to support mapping steps that can return many objects
         if (!context.useContextCollectionsOnly) {
-          context.collections = this.collectionsFromOptions.concat(context.collections || []);
+          context.collections = [...new Set(this.collectionsFromOptions.concat(context.collections || []))];
```

Comment:
Can you verify if this works in the least version of the supported 9.x server ? I believe spread operator may not work in those server versions.

Reply:
I'm 99% sure it works - it's used in this module too:

this.collectionsFromOptions = [...new Set(this.collectionsFromOptions)];

Note too that "duplicate collections" are technically not an issue - i.e. they have no adverse effect on ML functionality.

Reply:
@srinathgit It works. I tested it on 9.0-11. As @rjrudin there are other instances of the usage of spread operator and Set